zeus/monitor/energy.py

```diff
@@ -138,13 +147,23 @@ def __init__(
         self.approx_instant_energy = approx_instant_energy
 
         # Get gpus
-        self.gpus = get_gpus()
+        with contextlib.suppress(Exception):
+            self.gpus = get_gpus()
+
+        # Get cpus
+        with contextlib.suppress(Exception):
+            self.cpus = get_cpus()
```

Comment:
Okay. This is the worst possible way to handle no-GPU/no-CPU scenarios.

What if the user had two GPUs and somehow there actually **was** an error while initializing GPUs? What if the user's RAPL sysfs was somehow misconfigured and the user needs to figure out a fix for it? This will ignore all of those situations where the program ought to fail. It will allow the `ZeusMonitor` to execute in broken state. Also, if errors were actually suppressed, `self.gpus` or `self.cpus` will not exist in `ZeusMonitor` and accessing them will raise an `AttributeError` in the following lines.

Remember that erroring and dying (while providing informative error messages) is infinitely better than continuing execution in a broken state.

Reply:
Nevermind, pyright is complaining about function calls on a None. I think I would have to make "empty" GPUs and CPUs

Reply:
Added `EMPTYGPUs` and `EMPTYCPUs` to respective `common` files that raise `ValueError` for calls to any methods and `returns 0` for `__len__`